src/languages/typescript.ts

```diff
@@ -69,12 +73,50 @@ const getTags = (selection: SelectionWithEditor, node: SyntaxNode) => {
   return startTag != null && endTag != null ? startTag.concat(endTag) : null;
 };
 
-const findTypeNode = (node: SyntaxNode) => {
-  const typeAnnotationNode = node.children.find((child) =>
-    ["type_annotation", "opting_type_annotation"].includes(child.type)
-  );
-  return typeAnnotationNode?.lastChild ?? null;
-};
+function typeMatcher(): NodeMatcher {
+  const delimiterSelector = selectWithLeadingDelimiter(":");
+  return function (selection: SelectionWithEditor, node: SyntaxNode) {
+    if (
+      node.parent?.type === "new_expression" &&
+      node.type !== "new" &&
+      node.type !== "arguments"
+    ) {
+      const identifierNode = node.parent.children.find(
+        (n) => n.type === "identifier"
+      );
+      const argsNode = node.parent.children.find(
+        (n) => n.type === "type_arguments"
+      );
+      if (identifierNode && argsNode) {
```

Comment:
How come we require a type arguments node?  Shouldn't "take type" still work if you are just in `new Foo()`?  I think it is more consistent to just take the full constructor no matter what it is

Reply:
In that case it will do the same thing as `take token` which is fine. 
Done and done